src/rust/engine/fs/store/src/local.rs

```diff
@@ -308,11 +308,11 @@ impl UnderlyingByteStore for ShardedFSDB {
   }
 
   async fn remove(&self, fingerprint: Fingerprint) -> Result<bool, String> {
-    Ok(
-      tokio::fs::remove_file(self.get_path(fingerprint))
-        .await
-        .is_ok(),
-    )
+    let removed = tokio::fs::remove_file(self.get_path(fingerprint))
+      .await
+      .is_ok();
+    let _ = self.dest_initializer.lock().remove(&fingerprint);
```

Comment:
You should probably swap these two steps.

Assuming that the `dest_initializer` is process concurrency safe (...it might not be, but it actually needs to be...), then it would be ok to double-initialize a file.

Reply:
MMm good point